src/infix-abstract.ts

```diff
@@ -0,0 +1,69 @@
+
+const SUBSCRIPT_DIGITS = new Map(Object.entries({
+  "0": "₀",
+  "1": "₁",
+  "2": "₂",
+  "3": "₃",
+  "4": "₄",
+  "5": "₅",
+  "6": "₆",
+  "7": "₇",
+  "8": "₈",
+  "9": "₉",
+  "-": "₋"
+}));
+
+
+import { Notation } from "./notation";
+import Decimal from "break_infinity.js";
+
+export abstract class AbstractInfixNotation extends Notation {
+  public get name(): string {
+    return "Abstract Infix";
+  }
+
+  public formatDecimal(value: Decimal, places: number): string {
+    const p = Math.max(places, this.groupDigits-1);
+//     const offset = value.exponent % this.groupDigits;
+//     let exp = value.exponent - offset;
+    let exp = value.exponent;
+//     let mant = value.mantissa * Math.pow(10, offset);
+    let mant = value.mantissa;
+
+    let str = "";
+    // Loop through digits, from first to next-to-last
+    for (let minExp = exp - p; exp > minExp; exp--) {
+      const digit = Math.trunc(mant);
+      str += this.formatMantissa(digit);
+
+      // Output a separator on each group
+      if (!(exp % this.groupDigits))  {
+        str += this.formatExponent(exp);
+      }
+      mant = (mant - digit) * 10;
+    }
+
+    // Output last digit
+    str += this.formatMantissa(Math.round(mant));
+
+    // Only output separator at the end if it's the only one
+    if (p <= this.groupDigits && !(exp % this.groupDigits))  {
+      str += this.formatExponent(exp);
+    }
+
+    return str;
+  }
+
+  // Formats a small number into a subscript string
+  // Technically speaking, returns a string containing unicode sequences, each sequence
+  // representing one subscript digit of the given number, interpreted as base 10.
+  // "small" number means "smaller than the IEEE 754 mantissa limit", i.e. "smaller than 2^24"
+  protected numberToSubscript(n: number): string {
+    return n.toString(10).split("").map((d) => SUBSCRIPT_DIGITS.get(d)).join("");
```

Comment:
This library uses Decimal, and should work for mantissas up to 9e15. This function seems to do that, but the comment should change to something like
`// "small" number means "smaller than the Decimal mantissa limit", i.e. "smaller than 9e15"`

Reply:
Ok wait, *mantissa* limit? Didn't even know that existed TBH. The exponent limit is what I meant, and that's 9e15. The mantissa limit is usually 10 but some notations take it up to 1000 by using library internals (if your notation avoids that you can assume it's 10, though).

Reply:
Oh, I'm assuming that the `n` parameter is a `Number`, not a `Decimal`. This method is fed the `.exponent` or `.mantissa`

Thing is, the built-in `Number.prototype.toString()` method formats numbers depending on whether or not they can be represented with IEEE754 exponent of zero - for integers, numbers lower than the IEEE754 mantissa limit are displayed as digits only; otherwise one gets `.` and `e` as part of the string output.

I was thinking on doing hexadecimal output for this notation, but that seemed to be overkill.